rtloader/common/builtins/datadog_agent.c

```diff
@@ -653,3 +662,46 @@ static PyObject *set_external_tags(PyObject *self, PyObject *args)
     Py_RETURN_NONE;
 
 }
+
+/*! \fn PyObject *obfuscate_sql(PyObject *self, PyObject *args)
+    \brief This function implements the `datadog_agent.obfuscate_sql` method, obfuscating
+    the provided sql string.
+    \param self A PyObject* pointer to the `datadog_agent` module.
+    \param args A PyObject* pointer to a tuple containing the key to retrieve.
+    \return A PyObject* pointer to the value.
+
+    This function is callable as the `datadog_agent.obfuscate_sql` Python method and
+    uses the `cb_obfuscate_sql()` callback to retrieve the value from the agent
+    with CGO. If the callback has not been set `None` will be returned.
+*/
+static PyObject *obfuscate_sql(PyObject *self, PyObject *args)
+{
+    // callback must be set
+    if (cb_obfuscate_sql == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    PyGILState_STATE gstate = PyGILState_Ensure();
+
+    char *rawQuery;
+    if (!PyArg_ParseTuple(args, "s", &rawQuery)) {
+        return NULL;
+    }
+
+    char *obfQuery = NULL;
+    char *err = NULL;
+    obfQuery = cb_obfuscate_sql(rawQuery, &err);
+
+    PyObject *retval = NULL;
+    if (err != NULL) {
+        PyErr_SetString(PyExc_RuntimeError, err);
+        cgo_free(err);
+    }
+    if (obfQuery != NULL) {
```

Comment:
We should set an error in case both `err` and and `obfQuery` are null

Reply:
If they're both null why not just return null?

Reply:
Because `NULL` is not a PyObject (`PyNone` is the equivalent). But more importantly if both are null we should raise an error as the Go code most likely misbehaved. 

So either PyNone or and error. A user on the python side should not have to check for `None` but catch an exception, that's why I think we should raise one here.

Reply:
Good point. Updated to set an error in this case. Please take another look!